### PR TITLE
6.3 hardening day 3

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileService.java [hunk 1]
```diff
@@ -94,14 +89,6 @@ public void backup(String profileKey, Writer writer) {
     backuper.backup(profileKey, writer);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileService.java]: MINOR Remove this use of "backup"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileBackuper.java [hunk 2]
```diff
@@ -52,29 +51,33 @@
...
   public QProfileBackuper(QProfileReset reset, DbClient db) {
     this.reset = reset;
     this.db = db;
   }
 
+  /**
+   * @deprecated use {@link #backup(DbSession, QualityProfileDto, Writer)} instead
+   */
+  @Deprecated
   public void backup(String key, Writer writer) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileBackuper.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WebServiceEngine.java#L106](https://github.com/SonarSource/sonarqube/blob/c7c359ebf4f3561412de6967dbbfb5c1e003037f/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L106): NullPointerException might be thrown as 'action' is nullable here [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WebServiceEngine.java#L106](https://github.com/SonarSource/sonarqube/blob/e7c920a3f8110fb18fe3fa3a1184b4febb39a784/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L106): NullPointerException might be thrown as 'action' is nullable here [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WebServiceEngine.java#L106](https://github.com/SonarSource/sonarqube/blob/9ee8f05fd7c966ab9ba167fdacd332734c6c091c/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L106): NullPointerException might be thrown as 'action' is nullable here [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by julienlancelot** [comment]
Merged into feature/jl/6.3_hardening_4

**Pull request closed without merging**